implementation/src/main/java/io/smallrye/config/NameIterator.java

```diff
@@ -292,7 +278,7 @@ public boolean nextSegmentEquals(String other, int offs, int len) {
     }
 
     public String getNextSegment() {
-        return getNextSegment(acquireTmpBuilder()).toString();
+        return getNextSegment(new StringBuilder()).toString();
```

Comment:
Are we fine that we allocate a new one each time?

Reply:
Yes, we considerably reduced the use of the method with all of the other changes. The remaining uses are not in the hot path. I think we can get rid of it completely, but not a priority.

We still keep a `getNextSegment(StringBuilder b)` to use when required.